File: Modules/SEDO/EMS/SIE.Web.EMS/AssetDisposals/Commands/ImportAssetDisposalSparePartCommand.js
SIE.defineCommand('SIE.Web.EMS.AssetDisposals.Commands.ImportAssetDisposalSparePartCommand', {
    extend: 'SIE.Web.Common.Import.Commands.ImportExcelCommand',
    meta: { text: "导入", group: "business", iconCls: "icon-Upload icon-green" },
    onSuccessImported: function (me, res) {
        if (!me.view.getParent()) {
            me.view.reloadData();
        }
        else if (me._ownerView != null && me.view.getParent().getCurrent() != null) {
            me._ownerView.loadChildData(true)
        }
        Ext.MessageBox.close();
        if (res.Result.LoadSucessQty === 0 && res.Result.LoadFailQty === 0)
            res.Result.MessageList.push({ RowNum: 0, MsgType: 1, Message: "导入的文件为空！" })
        me.showInfoWin(res.Result);
    }
});

File: Modules/SEDO/EMS/SIE.Web.EMS/AssetDisposals/Commands/ImportAssetDisposalSparePartCommand.cs
using SIE.Common.Import;
using SIE.EMS.AssetDisposals;
using SIE.Web.Common.Import.Commands;
using System;
using System.Collections.Generic;
using System.Text;

namespace SIE.Web.EMS.AssetDisposals.Commands
{
    /// <summary>
    /// 导入设备回收
    /// </summary>
    public class ImportAssetDisposalSparePartCommand : ImportExcelCommand
    {
        /// <summary>
        /// 保存
        /// </summary>
        /// <param name="batch"></param>
        protected override void OnSave(IList<RowData> batch)
        {
            if (batch == null)
            {
                return;
            }
            var ImpMesResultList = RT.Service.Resolve<AssetDisposalController>().ImportOnSave(batch);
            importResult.MessageList.AddRange(ImpMesResultList);
        }
    }
}
